tools/cloud_functions/gcs_event_based_ingest/gcs_ocn_bq_ingest/main.py

```diff
@@ -426,19 +439,38 @@ def parse_notification(notification: dict) -> Tuple[str, str]:
     Args:
         notification(dict): Pub/Sub Storage Notification
         https://cloud.google.com/storage/docs/pubsub-notifications
+        Or Cloud Functions direct trigger
+        https://cloud.google.com/functions/docs/tutorials/storage
+        with notification schema
+        https://cloud.google.com/storage/docs/json_api/v1/objects#resource
     Returns:
         tuple of bucketId and objectId attributes
     Raises:
         KeyError if the input notification does not contain the expected
         attributes.
     """
-    try:
-        attributes = notification["attributes"]
-        return attributes["bucketId"], attributes["objectId"]
-    except KeyError:
-        raise RuntimeError(
-            "Issue with payload, did not contain expected attributes"
-            f"'bucketId' and 'objectId': {notification}") from KeyError
+    if {"bucket", "name", } <= notification.keys():
```

Comment:
Cleaner to check for:
notification.get("kind") == "storage#object"

Reply:
so much more explicit, love it.